scipy-1.0/minimize.tex

```diff
@@ -0,0 +1,118 @@
+\end{table}
+
+
+The \texttt{minimize} function provides methods for solving general non-linear optimization
+problems. It unifies several methods under the same general interface in order to facilitate
+interchanging between solvers with different characteristics.  Table~\ref{tab:minimize-unconstrained} summarizes
+all the methods available for solving unconstrained minimization problems of the type:
+\begin{equation}
+  \label{eq:unconstrained-minimization-prob}
+  \min_x f(x)
+\end{equation}
+where $f$ is a multivariate function $f: \mathbb{R}^n \rightarrow \mathbb{R}$ and $x \in \mathbb{R}^n$.
+
+There are a sufficient variety of methods to enable the user to solve problems with the diverse properties.
+Derivative-free methods (e.g. \texttt{Nelder-Mead} e \texttt{Powell}) are appropriate for minimizing noisy objective
+functions. When the derivatives are available, on the other hand, methods with faster convergence can also be used.
+
+There are algorithms implementing two competing strategies: line-search and trust-region. The so-called line-search (LS)
+methods choose a direction and then find the step length that gives the (approximately) optimal point in the given
+direction. Trust-region (TR) methods, on the other hand, fixes the maximum step length for which the local approximation
+is valid and finds approximately the best point inside this \textit{trust-region}.
+
+An example of trust-region method introduced in the latest releases is \texttt{trust-exact}, which a  provides a fast convergence rate by solving 
+ almost exactly the trust-region subproblem. Its downside is it also require the Hessian
+to be stored and factored every iteration, which may renders impossible the solution of
+very large problems.
+
+Some available unconstrained methods are appropriate for large-scale problems
+(i.e. problems with more than 10000 variables).
+\texttt{CG} method does not use second derivatives and is adequate to large-scale problems due to the low storage requirements.
+\texttt{Newton-CG}, \texttt{trust-ncg} and \texttt{trust-krylov} use the second order derivatives in order to get faster
+convergence but are still appropriate for large-scale problems because they factorate the Hessian in a inexact and iterative way,
+which does not require the Hessian to be actually stored.
+
+The method \texttt{trust-krylov}, introduced in the latest release, presents itself as a good compromise between
+\texttt{trust-ncg} and \texttt{trust-exact}: it provides an more accurate solution to the trust-region
+subproblem than \texttt{trust-ncg} but does not require the Hessian to be stored  and factored as in \texttt{trust-exact}.
+
+Besides unconstrained problems, the \texttt{minimize} function also provides methods for solving
+minimization problems subject to constraints. Table~\ref{tab:minimize-constrained} summarizes all
+constrained optimization methods available, with information of which type of
+constraints they are able to deal with.
+
```

Comment:
Do you intend to add more about these? This is a lot shorter than the part about the techniques for unconstrained problems.

Reply:
Ok, I decided to merge both tables into one and talk about constrained and unconstrained algorithms together.

It makes more sense.